src/routing/bootstrap.rs

```diff
@@ -190,6 +185,20 @@ impl<'a> State<'a> {
         while let Some((message, sender)) = self.recv_rx.next().await {
             match message.variant() {
                 Variant::BootstrapResponse(response) => {
+                    match response {
+                        BootstrapResponse::Rebootstrap(addrs) if addrs.is_empty() => {
+                            error!("Invalid Rebootstrap response: missing peers");
+                            continue;
+                        }
+                        BootstrapResponse::Join { elders_info, .. }
+                            if !elders_info.prefix.matches(&self.node.name()) =>
+                        {
+                            error!("Invalid Join response: bad prefix");
```

Comment:
this could happen when a node bootstrap to genesis which is just an adult of the sibling section of that node?
so that the genesis will repond with its own elders, and when the node try to rebootstrap to those elders, it will receive the proper list of elders of its own section ?

Reply:
Not sure I follow. When we send a `BootstrapRequest` to a section which doesn't match our name, we should receive back a `BootstrapResponse::Rebootstrap` with a new set of bootstrap contacts (to send another `BootstrapRequest` to). When we receive `BootstrapResponse::Join`, it should only be from a section that matches our name. If it isn't, then I think it's OK to ignore it.

> so that the genesis will repond with its own elders

Which it will do with a `BootstrapResponse::Rebootstrap`

> and when the node try to rebootstrap to those elders, it will receive the proper list of elders of its own section

Yes, but again, with a `BootstrapResponse::Rebootstrap`

Reply:
Ah, right, I messed up with BootstrapRequest and Join.
yeah, Join shall always for own section